connectors/src/lib/temporal_monitoring.ts

```diff
@@ -0,0 +1,89 @@
+import { Context } from "@temporalio/activity";
+import {
+  ActivityExecuteInput,
+  ActivityInboundCallsInterceptor,
+  Next,
+} from "@temporalio/worker";
+import StatsDClient from "hot-shots";
+
+import logger, { Logger } from "@connectors/logger/logger";
+
+import { WorkflowErrorType } from "./error";
+
+/** An Activity Context with an attached logger */
+export interface ContextWithLogger extends Context {
+  logger: typeof logger;
+}
+
+export const statsDClient = new StatsDClient({});
+
+export class ActivityInboundLogInterceptor
+  implements ActivityInboundCallsInterceptor
+{
+  public readonly logger: Logger;
+  private readonly context: Context;
+
+  constructor(ctx: Context, logger: Logger) {
+    this.context = ctx;
+    this.logger = logger.child({
+      activityName: ctx.info.activityType,
+      workflowName: ctx.info.workflowType,
+      workflowId: ctx.info.workflowExecution.workflowId,
+      workflowRunId: ctx.info.workflowExecution.runId,
+      activityId: ctx.info.activityId,
+    });
+
+    // Set a logger instance on the current Activity Context to provide
+    // contextual logging information to each log entry generated by the Activity.
+    (ctx as ContextWithLogger).logger = this.logger;
+  }
+
+  async execute(
+    input: ActivityExecuteInput,
+    next: Next<ActivityInboundCallsInterceptor, "execute">
+  ): Promise<unknown> {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let error: any = undefined;
+    const startTime = new Date();
+    const tags = [
+      `activity_name:${this.context.info.activityType}`,
+      `workflow_name:${this.context.info.workflowType}`,
+      `activity_id:${this.context.info.activityId}`,
+      `workflow_id:${this.context.info.workflowExecution.workflowId}`,
+      `workflow_run_id:${this.context.info.workflowExecution.runId}`,
+    ];
+
+    try {
+      return await next(input);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (err: any) {
+      error = err;
+      throw err;
+    } finally {
+      const durationMs = new Date().getTime() - startTime.getTime();
+      if (error) {
+        const errorType: WorkflowErrorType = error.type
```

Comment:
If an error comes from an internal library with a `type` field we'll be considering one of our own.

We probably need something a bit more magical:

```
let type = "unhandled_internal_activity_error";
if (error._is_dust_error) {
  type = error.type
}
```
Or a `isErr()` type pattern, both are plausible :+1:

Reply:
type guard FTW !